frontend/services/src/maps/mapsService.ts

```diff
@@ -0,0 +1,66 @@
+// src/services/googleMapsService.ts
+import { BasicData } from "@/components/events/create/forms/BasicForm";
+import { Environment, getEnvironment } from "@/utilities/environment";
+import { useLoadScript } from "@react-google-maps/api";
+
+const libraries: "places"[] = ["places"];
+
+const getGoogleMapsApiKey = () => {
+  const env = getEnvironment();
+  if (env === Environment.DEVELOPMENT) {
+    return process.env.GOOGLE_MAPS_DEV_API_KEY;
+  }
+  if (env === Environment.PRODUCTION) {
+    return process.env.GOOGLE_MAPS_PROD_API_KEY;
+  }
+  throw new Error("Google Maps API Key is not defined");
+};
+
+const loadGoogleMapsScript = () => {
+  const googleMapsApiKey = getGoogleMapsApiKey();
+  if (!googleMapsApiKey) {
+    return null;
+  }
+  return useLoadScript({
+    googleMapsApiKey: googleMapsApiKey,
+    libraries,
+  });
+};
+
+const initializeAutocomplete = (inputRef: React.RefObject<HTMLInputElement>, handlePlaceSelect: () => void) => {
+  if (!window.google || !window.google.maps) {
+    console.error("Google Maps JavaScript API is not loaded.");
+    return null;
+  }
+  if (!inputRef.current) {
+    return null;
+  }
+  const options = {
+    types: ["address"],
+    componentRestrictions: { country: "au" },
+  };
+  const autocomplete = new window.google.maps.places.Autocomplete(inputRef.current, options);
+  autocomplete.addListener("place_changed", handlePlaceSelect);
+  return autocomplete;
+};
+
+const validateLocation = (location: string, updateField: (fields: Partial<BasicData>) => void) => {
```

Comment:
this needs to return the lat long, it shouldn't have the updateField leak into the function